common/src/main/java/com/github/khanshoaib3/minecraft_access/features/read_crosshair/CrosshairNarrator.java

```diff
@@ -0,0 +1,31 @@
+package com.github.khanshoaib3.minecraft_access.features.read_crosshair;
+
+import net.minecraft.client.world.ClientWorld;
+import net.minecraft.util.hit.HitResult;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+public interface CrosshairNarrator {
+    @NotNull HitResult rayCast();
+
+    /**
+     * Determines weather the player is looking at the same block or entity.
+     * {@link #narrate(ClientWorld, boolean)} will only be called
+     * if this method returns a different non-null value than its previous call.
+     * A value of null indicates that the player is not currently looking at any block
+     * and thus {@link #narrate(ClientWorld, boolean)} should not be called.
+     * @param world The current world the player is in
+     * @param speakSide Weather the block side the player is pointing at should be narrated
+     * @return Any object comparable with {@link java.util.Objects#equals(Object, Object)} or null
+     */
+    @Nullable Object deduplication(@NotNull ClientWorld world, boolean speakSide, boolean speakConsecutiveBlocks);
```

Comment:
`deduplication` is not a clear name to explain its function, how about `getPlayerLookingAt`.
the `ClientWorld` param can be replaced by internal `WorldUtils.getClientWorld`.
(And... detailed method doc is good, but the explanation about Crosshair speaking isn't necessary here, this part can be moved to invocation of this method)

Reply:
The method's sole purpose is to indicate when the target changes and prevent unwanted duplicated narration (deduplication), it's equivalent to the old `query` variable. It only returns an Object that can be compared for changes, implementers can make it return anything that when changed indicates that the target has changed.

I just included the `world` param for convenience because I wasn't aware of `WorldUtils.getClientWorld` and didn't want to go through all the null checks neccessary to get it otherwise, I'll look into changing that in the morning.

I feel it's important to include the relationship between `deduplication` and `narrate` so that implementers (potentially in a future API) understand how it works without having to dig around the callsite.

Reply:
I know your purpose, but it doesn't fit current function signature: world, speakSide in, object out... Another choice is pass previous object into this method, then make it return a boolean value to indicate whether to speak.

Reply:
I've removed the `world` parameter, turns out I wasn't using it anymore anyway. I don't quite understand what you mean about the signature not matching the name, I think `deduplication` aptly describes the purpose of the method and is the most appropriate name.